prereise/gather/data/remap_ba_area/remap_ba_area.py

```diff
@@ -0,0 +1,285 @@
+import json
+import os
+import shutil
+
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import pandas as pd
+import requests
+
+from prereise.gather.data.remap_ba_area.map_data import (
+    miso_name_map,
+    states_to_map_with_counties,
+    swpp_name_map,
+)
+
+src = "ca_source"
+dst = "BA_map_name"
+missing = "missing"
+
+
+def get_full_filename(file_name):
+    return os.path.join(os.path.dirname(__file__), file_name)
+
+
+def compose_ba_map(output_shape_file="remap_ba_area.shp"):
+    """
+    Create the control area and subregion map. For CAs heavily overlapped, we define several zones and match the CAs to
+    these zones. The ca_source name is either mapped to one zone in BA_map_name, or allocated to multiple zones in
+    BA_map_name with fractions based on the area.
+    The function generates the remap_ba_area shape file and remap_ba_area.tif file.
+
+    :param str output_shape_file: output shape file name
+    :return: (*geopandas.DataFrame*)  -- the final control zone map
+
+    """
+
+    with open(get_full_filename("../BA_County_map.json")) as json_file:
+        ba2county_dict = json.load(json_file)
+    # Add two missing counties
+    # this may be fixed in future updates of BA_County_map.json
+    ba2county_dict["EPE"].append("Doña Ana__NM")
+    ba2county_dict["WAUE"].append("Oglala Lakota__SD")
+    # Turn dict into pandas table
+    county2ba_dict = {
+        county: ba
+        for (ba, ba_counties) in ba2county_dict.items()
+        for county in ba_counties
+    }
+    county2ba = pd.Series(county2ba_dict).to_frame().reset_index()
+    county2ba.columns = ["index", "BA"]
+    county2ba[["county", "state"]] = county2ba["index"].str.split(
+        "__", n=1, expand=True
+    )
+    county2ba = county2ba.drop(columns=["index"])
+    county2ba.set_index(["county", "state"], inplace=True)
+
+    # ercot_zones = (
+    #     pd.read_json(get_full_filename("ercot_zone_mapping.json"))
+    #     .rename({"County": "county", "Zone": "BA"}, axis="columns")
+    #     .assign(state="TX")
+    #     .set_index(["county", "state"])
+    # )
+    # county2ba.update(ercot_zones)
```

Comment:
If we don't need this code anymore, we should remove it fully.

Reply:
Deleted.